sqlglot/dialects/tsql.py

```diff
@@ -25,6 +42,14 @@ def _format_time(args):
     return _format_time
 
 
+def parse_date_delta(args, kind="ADD"):
+    unit = DATE_DELTA_INTERVAL.get(list_get(args, 0).text("this").lower(), "day")
+    if kind == "SUB":
+        return exp.DateDiff(this=list_get(args, 2), expression=list_get(args, 1), unit=unit)
+    else:
+        return exp.DateAdd(this=list_get(args, 2), expression=list_get(args, 1), unit=unit)
```

Comment:
Are we sure that calling a method on list_get(...) won't cause an exception? Also, maybe we can simplify this to:
```python
def parse_date_delta(args, kind="ADD"):
    unit = DATE_DELTA_INTERVAL.get(list_get(args, 0).name.lower(), "day")
    exp_class = exp.DateDiff if kind == "SUB" else exp.DateAdd
    return exp_class(this=list_get(args, 2), expression=list_get(args, 1), unit=unit)
```

Reply:
DATEDIFF(interval, date1, date2)
DATEADD (datepart , number , date )  

Both always should have 3 parameters (as far as I can tell from the docs) thus having it crash on an exception if list_get fails might not be the worst?

Reply:
I think it'd be better to notify the user that we're missing a third argument instead of throwing an AttributeError.

Reply:
Something like this? or should the error be thrown from a different location?
I see very little dialects implementing errors.
```
def parse_date_delta(exp_class):
    def inner_func(args):
        if not list_get(args, 2):
          raise IndexError(f"missing parameters for class: {exp_class}. 3 parameters are expected")
        unit = DATE_DELTA_INTERVAL.get(list_get(args, 0).name.lower(), "day")
        return exp_class(this=list_get(args, 2), expression=list_get(args, 1), unit=unit)

    return inner_func
```

Reply:
I think we can raise here. What do you say @tobymao?